### PR TITLE
Refactored private keys usage in consensus

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- refactored the private keys usage in consensus
  
## Proposed changes
- refactored the private keys usage in consensus

## Testing procedure
- N/A

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: consensus/spos/bls/subroundSignature.go [hunk 1]
```diff
@@ -82,7 +82,12 @@ func (sr *subroundSignature) doSignatureJob(_ context.Context) bool {
...
 
-		signatureShare, err := sr.SignatureHandler().CreateSignatureShare(sr.GetData(), uint16(selfIndex), sr.Header.GetEpoch())
+		signatureShare, err := sr.SignatureHandler().CreateSignatureShareUsingPublicKey(
+			sr.GetData(),
+			uint16(selfIndex),
+			sr.Header.GetEpoch(),
+			[]byte(sr.SelfPubKey()),
+		)
 		if err != nil {
 			log.Debug("doSignatureJob.CreateSignatureShare", "error", err.Error())
```
Comment (ssd04) [inline consensus/spos/bls/subroundSignature.go:92]: rename also this log entry

File: consensus/interface.go [hunk 2]
```diff
@@ -152,8 +152,8 @@ type ScheduledProcessor interface {
 // SignatureHandler defines the behaviour of a component that handles signatures in consensus
 type SignatureHandler interface {
 	Reset(pubKeys []string) error
-	CreateSignatureShare(msg []byte, index uint16, epoch uint32) ([]byte, error)
-	CreateSignatureShareWithPrivateKey(message []byte, index uint16, epoch uint32, privateKeyBytes []byte) ([]byte, error)
+	CreateSignatureShareUsingPublicKey(message []byte, index uint16, epoch uint32, publicKeyBytes []byte) ([]byte, error)
```
Comment (AdoAdoAdo) [inline consensus/interface.go:155]: can you rename to CreateSignatureShareForPublicKey and CreateSignatureForPublicKey ?

File: consensus/spos/bls/subroundSignature.go (hunk 1 above)
Comment (iulianpascalau) [reply consensus/spos/bls/subroundSignature.go:92]: done

File: consensus/interface.go (hunk 2 above)
Comment (iulianpascalau) [reply consensus/interface.go:155]: Renamed both methods to `CreateSignatureShareForPublicKey` and `CreateSignatureForPublicKey`

**Review by ssd04 — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**